java/ingest/src/main/java/sleeper/ingest/job/IngestJobQueueConsumer.java

```diff
@@ -215,22 +153,20 @@ public long ingest(IngestJob job, Schema schema, TableProperties tableProperties
                 job.getId(), keepAlivePeriod);
 
         // Run the ingest
-        List<FileInfo> ingestedFileInfoList = IngestRecordsUsingPropertiesSpecifiedMethod.ingestFromRecordIterator(
+        IngestJobRunner ingestJobRunner = new IngestJobRunner(
                 objectFactory,
-                stateStore,
                 instanceProperties,
                 tableProperties,
+                stateStoreProvider,
                 localDir,
                 s3AsyncClient,
                 hadoopConfiguration,
-                tableProperties.get(ITERATOR_CLASS_NAME),
-                tableProperties.get(ITERATOR_CONFIG),
-                concatenatingIterator);
-        long numRecordsWritten = ingestedFileInfoList.stream().mapToLong(FileInfo::getNumberOfRecords).sum();
+                schema);
```

Comment:
It's probably neater to make IngestJobRunner a field rather than make a new one each time ingest is called.

The table properties and schema can be loaded in IngestJobRunner.ingest, using a TablePropertiesProvider.